tests/integration/relations/pgbouncer_provider/application-charm/src/charm.py

```diff
@@ -0,0 +1,199 @@
+        """Event triggered when a database was created for this application."""
+        # Retrieve the credentials using the charm library.
+        logger.info(f"second database credentials: {event.username} {event.password}")
+        self.unit.status = ActiveStatus("received database credentials of the second database")
+
+    def _on_second_database_endpoints_changed(self, event: DatabaseEndpointsChangedEvent) -> None:
+        """Event triggered when the read/write endpoints of the database change."""
+        logger.info(f"second database endpoints have been changed to: {event.endpoints}")
+
+    # Multiple database clusters events observers.
+    def _on_cluster_database_created(self, event: DatabaseCreatedEvent) -> None:
+        """Event triggered when a database was created for this application."""
+        # Retrieve the credentials using the charm library.
+        logger.info(
+            f"cluster {event.relation.app.name} credentials: {event.username} {event.password}"
+        )
+        self.unit.status = ActiveStatus(
+            f"received database credentials for cluster {event.relation.app.name}"
+        )
+
+    def _on_cluster_endpoints_changed(self, event: DatabaseEndpointsChangedEvent) -> None:
+        """Event triggered when the read/write endpoints of the database change."""
+        logger.info(
+            f"cluster {event.relation.app.name} endpoints have been changed to: {event.endpoints}"
+        )
+
+    # Multiple database clusters events observers (for aliased clusters/relations).
+    def _on_cluster1_database_created(self, event: DatabaseCreatedEvent) -> None:
+        """Event triggered when a database was created for this application."""
+        # Retrieve the credentials using the charm library.
+        logger.info(f"cluster1 credentials: {event.username} {event.password}")
+        self.unit.status = ActiveStatus("received database credentials for cluster1")
+
+    def _on_cluster1_endpoints_changed(self, event: DatabaseEndpointsChangedEvent) -> None:
+        """Event triggered when the read/write endpoints of the database change."""
+        logger.info(f"cluster1 endpoints have been changed to: {event.endpoints}")
+
+    def _on_cluster2_database_created(self, event: DatabaseCreatedEvent) -> None:
+        """Event triggered when a database was created for this application."""
+        # Retrieve the credentials using the charm library.
+        logger.info(f"cluster2 credentials: {event.username} {event.password}")
+        self.unit.status = ActiveStatus("received database credentials for cluster2")
+
+    def _on_cluster2_endpoints_changed(self, event: DatabaseEndpointsChangedEvent) -> None:
+        """Event triggered when the read/write endpoints of the database change."""
+        logger.info(f"cluster2 endpoints have been changed to: {event.endpoints}")
+
+    def _on_run_sql_action(self, event: ActionEvent):
+        logger.error(event.params)
```

Comment:
The same for other `logger.error` calls on this file (the ones that are being used for debugging).
```suggestion
```